docs/BondingCurve.md

```diff
@@ -0,0 +1,170 @@
+
+Tokens issued via bonding curves can offer rights in the organization, such as **dividends** on future revenue or **governance rights** through token locking.
+
+This type of fundraising might allow for more flexibility, accountability, and alignment of incentives than alternative methods (such as ICOs or private fundraising).
+
+## How it works
+
+- Anyone can deposit a specified collateral token (such as DAI) to purchase bonded tokens.
+- The **bonding curve** is an automated market maker contract that mints tokens to buyers at an algorithmically determined price. The automated market maker allows users to buy or sell tokens at any time for known prices with minimal slippage.
+- When a user **buys** bonded tokens: Collateral tokens are split between the beneficiary treasury (to fund the organization) and the reserve (to facilitate liquidity for sells). The buyer receives bonded tokens based on the current price.
+
+![](./diagrams/out/bonding_curve_buy_flow.png)
+
+- When a user **sells** bonded tokens: The bonded tokens are burned and the seller receives collateral tokens based on the current price.
+
+![](./diagrams/out/bonding_curve_sell_flow.png)
+
+- When the beneficiary earns **revenue:** the income is split between direct income to the beneficiary treasury and **dividends** for the bonded token holders.
+
+![](./diagrams/out/bonding_curve_pay_flow.png)
+
+## Curve Economics
+
+- The specifics of curve design are an area of active research. A common general principle is to reward earlier contributors without inflating the buy price so high that it disincentives later participation.
+
+- Our current curve implementation is based on **bancor formula**, which allows for a wide variety of potential curves with simple parameters.
+  ![](./diagrams/out/bancor_curve_examples.png)
+
+[Bancor Whitepaper](https://storage.googleapis.com/website-bancor/2018/04/01ba8253-bancor_protocol_whitepaper_en.pdf)
+
+# Implementation
+
+Our initial bonding curve implementation supports:
+
+- Bancor-based curves.
+- Dividend distributions for bonded token holders.
+- A front-running guard via user-specified min and max prices.
+
+### Key Terms
+
+- **bondingCurve**: The ‘avatar’ of the bonding curve. It serves as the external interface to interact with the curve, with automated market maker and dividend tracking functions.
+- **bondedToken**: Token native to the curve. The bondingCurve contract has exclusive rights to mint and burn tokens.
+- **collateralToken**: Token accepted as collateral by the curve. (e.g. WETH or DAI)
```

Comment:
Are we supporting ETH as well?  Or only WETH?

Reply:
ETH - changed